### PR TITLE
Remove sharding rules

### PR DESCRIPTION
**Pull request by EinKrebs**

### CODE REVIEW

File: pkg/meta/meta.go [hunk 1]
```diff
@@ -55,57 +52,26 @@ func processDrop(ctx context.Context, dstmt spqrparser.Statement, isCascade bool
...
-			}
-		} else {
-			spqrlog.Zero.Debug().Str("rule", stmt.ID).Msg("parsed drop")
-			err := mngr.DropShardingRule(ctx, stmt.ID)
-			if err != nil {
-				return cli.ReportError(err)
-			}
-			return cli.DropShardingRule(ctx, stmt.ID)
-		}
+		return cli.ReportError(spqrerror.New(spqrerror.SPQR_INVALID_REQUEST, "sharding rules are removed from SPQR"))
```
Comment (Denchick) [inline pkg/meta/meta.go:55]: Maybe just create a well-known error in spqrerror package? For code-reuse purposes

File: pkg/meta/meta.go [hunk 2]
```diff
@@ -55,57 +52,26 @@ func processDrop(ctx context.Context, dstmt spqrparser.Statement, isCascade bool
...
-			}
 
 			krs, err = mngr.ListKeyRanges(ctx, stmt.ID)
 			if err != nil {
 				return err
 			}
 		}
 
-		if len(srs)+len(krs) != 0 && !isCascade {
+		if len(krs) != 0 && !isCascade {
```
Comment (Denchick) [inline pkg/meta/meta.go:74]: Maybe it is also good to count relations but im not sure

File: qdb/etcdqdb.go [hunk 3]
```diff
@@ -378,12 +378,6 @@ func (q *EtcdQDB) DropKeyRange(ctx context.Context, id string) error {
 	return err
 }
 
-// TODO : unit tests
-// TODO : implement
-func (q *EtcdQDB) MatchShardingRules(ctx context.Context, m func(shrules map[string]*ShardingRule) error) error {
```
Comment (Denchick) [inline qdb/etcdqdb.go:383]: Wait, why not drop *ShardingRules methods?

File: router/qrouter/proxy_routing.go [hunk 4]
```diff
@@ -606,20 +610,36 @@ func (qr *ProxyQrouter) deparseShardingMapping(
 var ParseError = fmt.Errorf("parsing stmt error")
 var ErrRuleIntersect = fmt.Errorf("sharding rule intersects with existing one")
 
-// TODO : unit tests
 // CheckTableIsRoutable Given table create statement, check if it is routable with some sharding rule
+// TODO : unit tests
 func (qr *ProxyQrouter) CheckTableIsRoutable(ctx context.Context, node *lyx.CreateTable, meta *RoutingMetadataContext) error {
+	ds, err := qr.mgr.GetRelationDistribution(ctx, node.TableName)
```
Comment (Denchick) [inline router/qrouter/proxy_routing.go:616]: @reshke could you review this method?

File: router/qrouter/proxy_routing_test.go [hunk 5]
```diff
@@ -851,23 +854,23 @@ func TestJoins(t *testing.T) {
...
-						ID:         "id2",
-						LowerBound: []byte("11"),
-					},
-				},
-				TargetSessionAttrs: "any",
-			},
-			err: qrouter.ComplexQuery,
-		},
+		//{
+		//	query: "SELECT * FROM xjoin JOIN yjoin on id=w_id where i = 15 ORDER BY id;",
```
Comment (Denchick) [inline router/qrouter/proxy_routing_test.go:858]: why?

File: test/feature/features/coordinator.feature [hunk 6]
```diff
@@ -129,21 +128,6 @@ Feature: Coordinator test
     UNREGISTER ROUTER r1;
     REGISTER ROUTER r1 ADDRESS regress_router::7000
     """
-    When I run SQL on host "router-admin"
```
Comment (Denchick) [inline test/feature/features/coordinator.feature:132]: Is there any ways to see list of relation and columns?

File: test/regress/tests/router/expected/joins.out [hunk 7]
```diff
@@ -81,8 +69,21 @@ NOTICE: send query to shard(s) : sh1,sh2
...
 -- result is not full
 --SELECT * FROM xjoin JOIN yjoin on true ORDER BY id;
+SELECT * FROM xjoin JOIN yjoin on id=w_id where yjoin.w_id = 15 ORDER BY id;
+NOTICE: send query to shard(s) : sh2
+ id | w_id 
+----+------
+ 15 |   15
+(1 row)
+
+-- TODO: complex query | figure a way to route this to specific shard
```
Comment (Denchick) [inline test/regress/tests/router/expected/joins.out:79]: @reshke any ideas?

File: test/feature/features/memqdb.feature [hunk 8]
```diff
@@ -1,175 +1,6 @@
 Feature: MemQDB save state into a file
-
-  Scenario: Sharding rules restored
```
Comment (Denchick) [inline test/feature/features/memqdb.feature:3]: It is really sad to drop these tests without writing new ones in this PR

**Review by Denchick — CHANGES_REQUESTED** [review]

File: cmd/mover/main.go [hunk 9]
```diff
@@ -146,6 +143,15 @@ WHERE column_name=$1;
 func main() {
 	flag.Parse()
 
+	if *lb != "" {
```
[the remark this answers is not in the document]
Comment (reshke) [reply cmd/mover/main.go:146]: tests use mover?

File: coordinator/provider/coordinator.go [hunk 10]
```diff
@@ -436,21 +436,6 @@ func (qc *qdbCoordinator) AddRouter(ctx context.Context, router *topology.Router
...
-	shRules := make([]*shrule.ShardingRule, 0, len(rulesList))
-	for _, rule := range rulesList {
-		shRules = append(shRules, shrule.ShardingRuleFromDB(rule))
-	}
-
-	return shRules, nil
-}
-
 // TODO : unit tests
 func (qc *qdbCoordinator) ListShardingRules(ctx context.Context, distribution string) ([]*shrule.ShardingRule, error) {
```
[the remark this answers is not in the document]
Comment (reshke) [reply coordinator/provider/coordinator.go:440]: no, postpone

File: test/feature/features/coordinator.feature (hunk 6 above)
Comment (EinKrebs) [reply test/feature/features/coordinator.feature:132]: There aren't now, I suggest adding more output to `SHOW distributions` command in separate PR

**Review by reshke — APPROVED** [review]

**Review by Denchick — COMMENTED** [review]
Discussed to do in next PRs:

- Drop `*ShardingRules` in QDB
- Add the test "it is not possible to attach the same table to dataspace"
- Add test "restore relations from memqdb file" [here](https://github.com/pg-sharding/spqr/pull/510/files#diff-b8d80a44f1cd5ec7a637f9b12ce6cc985de07e1c09107fc067d5776b583591f4R32)
- #512

**Review by Denchick — APPROVED** [review]

**Pull request merged**